discord/poll.py

```diff
@@ -0,0 +1,641 @@
+
+        # NOTE: These attributes are set manually when calling
+        # _from_data, so it should be ``None`` now.
+        self._message: Optional[Message] = None
+        self._state: Optional[ConnectionState] = None
+        self._results: Optional[PollResultPayload] = None
+        self._expiry: Optional[datetime.datetime] = None
+
+    def _update(self, message: Message) -> None:
+        self._state = message._state
+        self._message = message
+
+        if not message.poll:
+            # Something should go very wrong for this if block to be called
+            return
+        # These attributes are accessed via message.poll as there
+        # is where all the data is stored
+        self._expiry = message.poll.expiry
+        self._results = message.poll._results
+
+        if not self._results:
+            return
+
+        for count in self._results['answer_counts']:
+            answer = self.get_answer(int(count['id']))
+            if not answer:
+                continue
+            answer._update_with_results(count)
+
+    def _handle_vote(self, answer_id: int, added: bool, self_voted: bool = False):
+        answer = self.get_answer(answer_id)
+        if not answer:
+            return
+
+        answer._handle_vote_event(added)
+        answer.self_voted = self_voted
+
+    @classmethod
+    def _from_data(cls, *, data: PollPayload, message: Message, state: ConnectionState) -> Self:
+        multiselect = data.get('allow_multiselect', False)
+        layout_type = try_enum(PollLayoutType, data.get('layout_type', 1))
+        question_data = data.get('question')
+        question = question_data.get('text')
+        expiry = utils.parse_time(data['expiry'])  # If obtained via API, then expiry is set.
+        duration = expiry - message.created_at
+        # self.created_at = message.created_at
+        # duration = self.created_at - expiry
+
+        if (duration.total_seconds() / 3600) > 168:  # As the duration may exceed little milliseconds then we fix it
+            duration = datetime.timedelta(days=7)
+
+        self = cls(
+            duration=duration,
+            multiselect=multiselect,
+            layout_type=layout_type,
+            question=question,
+        )
+        self._answers = [PollAnswer(data=answer, message=message, poll=self) for answer in data['answers']]
+        self._message = message
+        self._state = state
+        self._expiry = expiry
+
+        try:
+            self._results = data['results']
+        except KeyError:
+            self._results = None
+
+        return self
+
+    def _to_dict(self) -> PollCreatePayload:
+        data: PollCreatePayload = {
+            'allow_multiselect': self.multiselect,
+            'question': self._question_media.to_dict(),
+            'duration': self._hours_duration,
+            'layout_type': self.layout_type.value,
+            'answers': [{'poll_media': answer._to_dict()} for answer in self.answers],
```

Comment:
```suggestion
            'answers': [answer.to_dict() for answer in self.answers],
```

If you take my above suggestion, this can be simplified.